groups/bsl/bsls/bsls_timeutil.cpp

```diff
@@ -448,114 +448,57 @@ struct MachTimerUtil {
 
   private:
```

Comment:
This is fairly extensive.   Looking at it I think it's fine, but needs a few changes (which i can't comment on directly because of github):

1. I suspect the include of <mach/mach_time.h> is no longer the right include... it looks like it should just be <time.h>?

2. Please change "MachTimerUtil" to "DarwinTimerUtil".
3. 
It seems based on documentation that this should also work on non-m1 darwin machines, but we'll have to try to verify that.